### ISSUE TITLE
Does the Notary server need quotas?

### ISSUE THREAD

**Issue by stefanberger** [theupdateframework/notary#1265]
I am wondering whether the Notary server should check how many entries a users has in the database and reject new entries once a certain limit has been reached so that filling up the database with useless data can be avoided?

**Comment by endophage** [comment]
I don't know how I feel about that... It's not a great situation if a user suddenly can't push an update when they need to fix a vuln that's live in a package signed in to a notary repo. Additionally, a quota based on number of records would greatly penalize people with repos that only have a small number of targets.

I'd be more inclined to add rate limits for now and wait until somebody is complaining about their database filling up to even think about quotas. If notary is deployed with token auth, it would be a trusted user filling up the database, and if their data is legitimate, quotas are a counterproductive measure.

**Comment by HuKeping** [comment]
With the [optional auth section](https://github.com/theupdateframework/notary/blob/master/docs/reference/server-config.md#auth-section-optional), unauthorized data will not get into the database.

**Comment by stefanberger** [comment]
@HuKeping True, it's not all open and at least makes it easier find out who did it.
